Troubleshooting.md

```diff
@@ -548,7 +549,68 @@ If you get this error
+  File "/home/pratiksolanki/.pyenv/versions/3.8.15/lib/python3.8/contextlib.py", line 425, in enter_context
+    result = _cm_type.__enter__(cm)
+  File "/home/pratiksolanki/.pyenv/versions/3.8.15/lib/python3.8/contextlib.py", line 113, in __enter__
+    return next(self.gen)
+  File "/home/pratiksolanki/opensource/oppia/scripts/servers.py", line 461, in managed_ng_build
+    yield proc
+  File "/home/pratiksolanki/.pyenv/versions/3.8.15/lib/python3.8/contextlib.py", line 525, in __exit__
+    raise exc_details[1]
+  File "/home/pratiksolanki/.pyenv/versions/3.8.15/lib/python3.8/contextlib.py", line 510, in __exit__
+    if cb(*exc_details):
+  File "/home/pratiksolanki/.pyenv/versions/3.8.15/lib/python3.8/contextlib.py", line 131, in __exit__
+    self.gen.throw(type, value, traceback)
+  File "/home/pratiksolanki/opensource/oppia/scripts/servers.py", line 138, in managed_process
+    raise Exception(
+Exception: Process Angular Compiler(name="sh", pid=2350) exited unexpectedly with exit code 137
+
+During handling of the above exception, another exception occurred:
+
+Traceback (most recent call last):
+  File "/home/pratiksolanki/.pyenv/versions/3.8.15/lib/python3.8/runpy.py", line 194, in _run_module_as_main
+    return _run_code(code, main_globals, None,
+  File "/home/pratiksolanki/.pyenv/versions/3.8.15/lib/python3.8/runpy.py", line 87, in _run_code
+    exec(code, run_globals)
+  File "/home/pratiksolanki/opensource/oppia/scripts/start.py", line 233, in <module>
+    main()
+  File "/home/pratiksolanki/opensource/oppia/scripts/start.py", line 229, in main
+    dev_appserver.wait()
+  File "/home/pratiksolanki/.pyenv/versions/3.8.15/lib/python3.8/contextlib.py", line 525, in __exit__
+    raise exc_details[1]
+  File "/home/pratiksolanki/.pyenv/versions/3.8.15/lib/python3.8/contextlib.py", line 510, in __exit__
+    if cb(*exc_details):
+  File "/home/pratiksolanki/.pyenv/versions/3.8.15/lib/python3.8/contextlib.py", line 382, in _exit_wrapper
+    callback(*args, **kwds)
+  File "/home/pratiksolanki/opensource/oppia/scripts/start.py", line 132, in call_extend_index_yaml
+    extend_index_yaml.main()
+  File "/home/pratiksolanki/opensource/oppia/scripts/extend_index_yaml.py", line 92, in main
+    with open(WEB_INF_INDEX_XML_PATH, 'r', encoding='utf-8') as f:
+FileNotFoundError: [Errno 2] No such file or directory: '/home/pratiksolanki/opensource/oppia/../cloud_datastore_emulator_cache/WEB-INF/appengine-generated/datastore-indexes-auto.xml'
+```
+
+The error code 137 indicates that the Angular compiler ran out of memory. This happens because by default, WSL2 only gives the host operating system (Ubuntu in our case) 50% of the machine's RAM. If you have 8 GB RAM, for example, this means Ubuntu only gets 4 GB, which is not enough to run Oppia. To fix this, you need to follow [Microsoft's instructions](https://learn.microsoft.com/en-us/windows/wsl/wsl-config) to increase the amount of RAM given to the host operating system.
```

Comment:
```suggestion
The error code 137 indicates that the Angular compiler ran out of memory. This happens because WSL2 only gives the host operating system (Ubuntu, in our case) 50% of the machine's RAM by default. If you have 8 GB RAM, for example, Ubuntu only gets 4 GB, which is insufficient to run Oppia. To fix this, you need to follow [Microsoft's instructions](https://learn.microsoft.com/en-us/windows/wsl/wsl-config) to increase the amount of RAM given to the host operating system.
```

Reply:
done